src/test/java/com/mentoring/framework/utils/EmailUtils.java

```diff
@@ -25,12 +25,28 @@
     private static final String USERNAME = "learn.ta.2019@gmail.com";
     private static final String PASSWORD = "Test1234!";
 
-    public static List<String> checkEmailsForIpChanges() {
-        return getListOfEmailContent(HOST, MAIL_STORE_TYPE, USERNAME, PASSWORD);
+    public static String checkLastEmailForIpChanges() {
+        return getLastEmailContent(HOST, MAIL_STORE_TYPE, USERNAME, PASSWORD);
     }
 
-    private static List<String> getListOfEmailContent(String host, String storeType, String user,
-                                                      String password) {
+    public static String getIpAddressFromEmail() {
+        String emailContent = checkLastEmailForIpChanges();
+        String ipAddress = "";
+
+        // find the ip address right after 'SkyXplore: http://'
+        Pattern pattern = Pattern.compile("(?<=SkyXplore: )http(s)*:\\/\\/(\\d{2,3}\\.){3}\\d{2,3}");
```

Comment:
This won't find IP address with 1 digit block in it, please fix it. Otherwise it's a neat regex!
A note though: you don't really need SkyXplore in the regex. IP address is the same for all the apps and we store the ports in environment config files.

Reply:
1 digit block: Sure, you are right.
SkyXplore: later we should use the port number as well from the e-mail.